### DISCUSSION TITLE
Should the emissions test button work now?

### DISCUSSION THREAD

**Discussion by wbradmoore** [alandtse/tesla#472] [Q&A]
I saw it in the release notes for 3.9.0 but dont see any such button in Home Assistant after updating and reloading - am I missing something?

**Comment by InTheDaylight14** [comment]
If your vehicle does not have a pedestrian speaker (or if it ran into problems setting up the entity) then the button is disabled by default. You can go into the integration entity list and see if the entity has been disabled. 

If you don't see the entity remove the integration and re-add it. 

My M3 does not have the pedestrian speaker. When writing the code I was not able to test if a sound is actually made but it _should_ work just fine.

**Reply by InTheDaylight14** [reply]
Also, make sure you have v3.9.1 installed. It had a fix for the button
